src/chordsymbol.ts

```diff
@@ -395,227 +222,123 @@ export class ChordSymbol extends Modifier {
-    }
-
-    // For superscripts that follow a letter without much top part, move it to the left slightly
-    let preKernUpper = false;
-    let preKernLower = false;
-    if (prevSymbol !== undefined && prevSymbol.symbolType === SymbolTypes.TEXT) {
-      preKernUpper = ChordSymbol.upperKerningText.some((xx) => xx === prevSymbol.text[prevSymbol.text.length - 1]);
-      preKernLower = ChordSymbol.lowerKerningText.some((xx) => xx === prevSymbol.text[prevSymbol.text.length - 1]);
-    }
-
-    const kerningOffsetPixels = ChordSymbol.kerningOffset * this.textFormatter.fontSizeInPixels;
-    // TODO: adjust kern for font size.
-    // Where should this constant live?
-    if (preKernUpper && currSymbol.symbolModifier === SymbolModifiers.SUPERSCRIPT) {
-      adjustment += kerningOffsetPixels;
-    }
-
-    if (preKernLower && currSymbol.symbolType === SymbolTypes.TEXT) {
-      if (currSymbol.text[0] >= 'a' && currSymbol.text[0] <= 'z') {
-        adjustment += kerningOffsetPixels / 2;
-      }
-      if (ChordSymbol.upperKerningText.some((xx) => xx === prevSymbol?.text[prevSymbol.text.length - 1])) {
-        adjustment += kerningOffsetPixels / 2;
-      }
-    }
-    return adjustment;
+    return ChordSymbol.subscriptOffset * Font.convertSizeToPixelValue(this.textFont?.size);
   }
 
   /**
    * ChordSymbol allows multiple blocks so we can mix glyphs and font text.
    * Each block can have its own vertical orientation.
    */
-  // eslint-disable-next-line
-  getSymbolBlock(params: any = {}): ChordSymbolBlock {
-    const symbolType = params.symbolType ?? SymbolTypes.TEXT;
-    const symbolBlock: ChordSymbolBlock = {
-      text: params.text ?? '',
-      symbolType,
-      symbolModifier: params.symbolModifier ?? SymbolModifiers.NONE,
-      xShift: 0,
-      yShift: 0,
-      vAlign: false,
-      width: 0,
-    };
-
-    // Note: symbol widths are resolution and font-independent.
-    // We convert to pixel values when we know what the font is.
-    if (symbolType === SymbolTypes.GLYPH && typeof params.glyph === 'string') {
-      const glyphArgs = ChordSymbol.glyphs[params.glyph];
-      const glyphPoints = 20;
-      symbolBlock.glyph = new Glyph(glyphArgs.code, glyphPoints, { category: 'chordSymbol' });
-    } else if (symbolType === SymbolTypes.TEXT) {
-      symbolBlock.width = this.textFormatter.getWidthForTextInEm(symbolBlock.text);
-    } else if (symbolType === SymbolTypes.LINE) {
-      symbolBlock.width = params.width;
+  getSymbolBlock(
+    params: Partial<{
+      text: string;
+      symbolModifier: SymbolModifiers;
+    }> = {}
+  ): ChordSymbolBlock {
+    const symbolBlock = new ChordSymbolBlock(
+      params.text ?? '',
+      params.symbolModifier ?? SymbolModifiers.NONE,
+      0,
+      0,
+      false
+    );
+
+    if (symbolBlock.isSubscript()) symbolBlock.setYShift(this.subscriptOffset);
+    if (symbolBlock.isSuperscript()) symbolBlock.setYShift(this.superscriptOffset);
```

Comment:
In general I am not a fan of single line if statements. I'd prefer to always use braces even if it's only one statement:

```js
if (condition) {
    <a single line of code>
}
```

It sets us up for the future and prevents possible bugs (in the future).

Reply:
done